tests/e2e/specs/wcpay/shopper/shopper-klarna-checkout.spec.js

```diff
@@ -44,16 +41,16 @@ describe( 'Klarna checkout', () => {
 			'#payment-method-message iframe'
 		);
 		const paymentMethodMessageIframe = await paymentMethodMessageFrameHandle.contentFrame();
-		const productMessaging = await paymentMethodMessageIframe.waitForSelector(
-			'button[aria-label="Open Learn More Modal"]'
-		);
-		// scrolling vertically to get the "product messaging" into view, since it seems that otherwise clicking on it fails.
-		await page.evaluate( () => {
-			window.scrollBy( 0, 100 );
-		} );
-		await productMessaging.click();
 
-		// we need to wait for the iframe to be added by Stripe JS after clicking on the element.
+		// Click on Klarna link to open the modal.
+		await paymentMethodMessageIframe.evaluate( ( selector ) => {
+			const element = document.querySelector( selector );
+			if ( element ) {
+				element.click();
+			}
+		}, 'button[aria-label="Open Learn More Modal"]' );
```

Comment:
Is this `page.evaluate()` done to avoid the `window.scrollBy()`, which could be flakey?

Reply:
Yes. If we use puppeteer methods in this case, it may take a slight delay for the iframe to load the content to DOM. With evaluate, we don't have to wait and click the button immediately.